### DISCUSSION TITLE
Migration of Web3Auth Testnet to Goerli

### DISCUSSION THREAD

**Discussion by yashovardhan** [Web3Auth/Web3Auth#911] [Announcements]
Hey Web3Auth Community,

Due to the recent change in the Ethereum testnets and the support deprecation by our node providers, we're moving our `testnet` from `Ropsten` to **`Goerli`**. 

`mainnet` & `cyan` is not affected by this change, things should be working as expected.

For people using Web3Auth Plug and Play SDKs, this should not cause any major change in the functionality. However, with Self Host SDKs (tKey & Custom Auth Web and Mobile SDKs), **please update to the latest SDK**.

If you're passing a custom URL in the `networkUrl`, please swap it to any Goerli Testnet URL (if you want to use a custom url), after updating to latest SDK.  Ideally, you can also just switch to use the `network` field as `testnet` directly and delete the `networkUrl` field altogether. 

### CustomAuth Web SDK

```js
import TorusSdk from "@toruslabs/customauth";
 
const torusdirectsdk = new TorusSdk({
 baseUrl: `${window.location.origin}/serviceworker`,
 enableLogging: true,
 network: "testnet",
});
```

### CustomAuth Android SDK
```java
CustomAuthArgs args = new CustomAuthArgs("https://scripts.toruswallet.io/redirect.html", TorusNetwork.TESTNET, "torusapp://org.torusresearch.customauthandroid/redirect");
```

### CustomAuth iOS SDK
```swift
CustomAuth(aggregateVerifierType: .singleLogin, aggregateVerifierName: "torus-auth0-github-lrc", subVerifierDetails: [sub], network: .TESTNET)
```

### If you are not able to see your verifiers on dashboard you can try either of the three solutions given below:-
- Login on [Web3Auth Dashboard](https://dashboard.web3auth.io/) in a new browser profile.
- Go to [Torus Wallet](https://app.tor.us/) and clear the local storage in your browser developer console and login to [Web3Auth Dashboard](https://dashboard.web3auth.io/)
- Login to [Web3Auth Dashboard](https://dashboard.web3auth.io/) , open the torus wallet from bottom left wallet icon and change the network from `ropsten`/`goerli` to `mainnet` from wallet setting and change it back to `goerli`.

Either of the above three should be able to show you your old verifiers if not showing now.

**Comment by parth7326** [comment]
Couldn't get it to work even after making these changes, we're using self hosting solution and it is giving this error on trigger login
`
Error: Returned values aren't valid, did it run Out of Gas? You might also see this error if you are not using the correct ABI for the contract you are retrieving data from, requesting data from a block number that does not exist, or querying a node which is not fully synced.
    at ABICoder.decodeParametersWith (index.js:304:1)
    at ABICoder.decodeParameters (index.js:291:1)
    at Contract._decodeMethodReturn (index.js:494:1)
    at Method.outputFormatter (index.js:801:1)
    at Method.formatOutput (index.js:148:1)
    at sendTxCallback (index.js:536:1)
    at index.js:305:1
    at index.js:124:1`

**Reply by vinayakrb** [reply]
Facing the same error when trying to use self host solution. Any update on this?

**Reply by armanvaziri** [reply]
Also facing this error when using the self host solution. Update please.

**Reply by yashovardhan** [reply]
This error happens when your contract is deployed on a different network than the one you're trying to approach. For Web3Auth this might mean one of the two things:
- You're pointing to the Ropsten contract but using Goerli RPC network
- You're pointing to Goerli contract but pointing to Ropsten RPC network

Our latest SDKs are pointing to Goerli and if you just change the parameters to testnet and **not** of pass the `networkUrl`, it should fine. Ideally don't hard code the proxy address, change it to the one in test folder of the repo.

Please update to the latest SDK and let us know if you're still facing the error.
